partner_firstname/views/res_partner.xml

```diff
@@ -0,0 +1,79 @@
+<odoo>
+    <record id="view_partner_form" model="ir.ui.view">
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form" />
+        <field name="priority" eval="99" />
+        <field name="arch" type="xml">
+                <xpath
+                expr="//div[hasclass('oe_title')]//field[@name='name'][1]"
```

Comment:
Replacing and using [1] are both bad practices in general. 
Here the problem is the core view added now 2 times the name, with id individual and company. 
The point is there updating both, not removing one. 
This is needed to set the field as readonly, so that when name is given a value through the computed field as soon as you are changing the firstname or lastname, it doesn't arrive to the create, that would call to the reverse function ending up in a wrong assignment of firstname and lastname. 

<xpath expr="//field[@id='individual']" position="attributes">
                    <attribute name="attrs">
                        {
                        'readonly': [('is_company', '=', False)],
                        'required': [('type', '=', 'contact'), ('is_company', '=', True)]
                        }
                    </attribute>
                </xpath>
                <xpath expr="//field[@id='company']" position="attributes">
                    <attribute name="attrs">
                        {
                        'readonly': [('is_company', '=', False)],
                        'required': [('type', '=', 'contact'), ('is_company', '=', True)]
                        }
                    </attribute>
                </xpath>

Reply:
The original form in Odoo 15 consists of 2 name fields, one for companies and one for individuals, with the only difference : the placeholder. As we don't need the individual name to be displayed anymore, I replaced it. The firstname and lastname will appear for individuals, the name field will appear for companies. In any case, the name field will still be available on the form (just invisible), I don't see why the inverse function would go wrong? The tests are running fine.

Please suggest another solution here.

` <field id="company" name="name" default_focus="1" placeholder="e.g. Lumber Inc" attrs="{'required' : [('type', '=', 'contact')], 'invisible': [('is_company','=', False)]}"/>
                            <field id="individual" name="name" default_focus="1" placeholder="e.g. Brandom Freeman" attrs="{'required' : [('type', '=', 'contact')], 'invisible': [('is_company','=', True)]}"/>
                     `

Reply:
Sorry I forgot to put it as code. I had already given you the solution in my previous input.  
That's needed in both views. And tests will run
For individuals you also need the name, it is computed from the firstname and lastname, that's how this module was working in previous versions. 
If you have some name field as not readonly, as soon as you changed firstname/lastname, the name is computed and gets a  value. When you save, it comes as a value, and then in the create it tries to split it with the same function as the inverse to get the firstname and lastname, assigning them in the wrong order if you for instance just gave firstname and not lastname.

Reply:
Thanks for the explanation

Reply:
I realized I had a mistake in my proposal. This would be the right one, as otherwise the invisible part from the core was broken
`<xpath expr="//field[@id='individual']" position="attributes">
                    <attribute name="attrs">
                        {
                        'readonly': [('is_company', '=', False)],
                        'required': [('type', '=', 'contact'), ('is_company', '=', False)],
                        'invisible': [('is_company','=', True)]
                        }
                    </attribute>
                </xpath>
                <xpath expr="//field[@id='company']" position="attributes">
                    <attribute name="attrs">
                        {
                        'readonly': [('is_company', '=', False)],
                        'required': [('type', '=', 'contact'), ('is_company', '=', True)],
                        'invisible': [('is_company','=', False)]
                        }
                    </attribute>
                </xpath>`

Reply:
Already did that. I noticed the invisible attrs was missing in your proposal.